### ISSUE TITLE
Incorrect information regarding Virtual Tables by SQL Views

### ISSUE THREAD

**Issue by Squire4Hire** [MicrosoftDocs/powerapps-docs#4893]
Please update this information to reflect that GUIDs are not accessible as Primary Fields when the Virtual Table is created from a SQL View. And as such, Virtual Tables that come from SQL Views will not work correctly as lookup columns.

Thank you.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 5ecd34d0-7724-9d0b-025b-63eeadb7b32a
* Version Independent ID: 5ecd34d0-7724-9d0b-025b-63eeadb7b32a
* Content: [Limitations and troubleshooting virtual tables with Power Apps - Power Apps](https://learn.microsoft.com/en-us/power-apps/maker/data-platform/limits-tshoot-virtual-tables?tabs=sql)
* Content Source: [powerapps-docs/maker/data-platform/limits-tshoot-virtual-tables.md](https://github.com/MicrosoftDocs/powerapps-docs/blob/main/powerapps-docs/maker/data-platform/limits-tshoot-virtual-tables.md)
* Service: **powerapps**
* Sub-service: **dataverse-maker**
* GitHub Login: @NHelgren
* Microsoft Alias: **nhelgren**

**Comment by bishalgoswami** [comment]
Thanks for your comment, @Squire4Hire. This issue has been assigned to the doc author for review.

**Comment by Squire4Hire** [comment]
I want to bump this.

I did more testing yesterday to see if anything has changed. It appears to me that this issue isn't limited to SQL Views. Virtual Tables created directly from SQL tables straight up don't work in Dataverse the way MS suggests. Sure, you can hook them up but you will not see lookup values displayed when used as a lookup in a Dataverse table.

The value exists, but the Primary Field is not displayed:
![image](https://github.com/MicrosoftDocs/powerapps-docs/assets/5404664/1bd6c2f3-a436-4761-96e8-3312ceddc337)

Unless I'm missing something, there is nothing in the documentation that suggests this shouldn't be possible.

**Comment by NHelgren** [comment]
Hi @Squire4Hire, All virtual tables display the Primary Name field as the lookup value, this cannot be changed. additional values an be retrieved though use of Automate or formulas in canvas apps if needed for the apps.

**Comment by Squire4Hire** [comment]
> Hi @Squire4Hire, All virtual tables display the Primary Name field as the lookup value, this cannot be changed. additional values an be retrieved though use of Automate or formulas in canvas apps if needed for the apps.
 
Perhaps I haven't made myself clear enough. Let me try again...
 
I am trying to use a Virtual Table as a _Lookup Value_ for an existing Dataverse table. This requires a virtual table to be configured in PowerApps as an added Virtual Table from a SQL Server datasource.
 
**Problem 1:** It is not possible to select a GUID key column if the table is created from a SQL View. This is contrary to what the documentation suggests is possible. SQL Views _cannot_ be used as Virtual Tables if a GUID field cannot be selected at the time of virtual table creation.

![image](https://github.com/MicrosoftDocs/powerapps-docs/assets/5404664/30f1bf7c-f590-4606-a84e-3de3f55c6107)

**Problem 2:** If the Virtual Table is created via a direct link to the original SQL Table, users are not able to use this Virtual Table as a _Lookup Field_ on another Dataverse table as the Primary Value field is not displayed correctly in the Parent Table. Yes, the GUID value exists when selected. Yes, the list of items appears in the dropdown when a user makes a selection. But the Primary Value disappears once the screen is reloaded, or when it is presented on any other Dataverse Form or View.

![image](https://github.com/MicrosoftDocs/powerapps-docs/assets/5404664/6f1369f4-7dc9-467a-b470-0e1015bd2af0)
![image](https://github.com/MicrosoftDocs/powerapps-docs/assets/5404664/96dbff21-dc57-4f9e-b46f-5bd0319aac31)
![image](https://github.com/MicrosoftDocs/powerapps-docs/assets/5404664/f3beca9d-e1f7-43ca-9bee-100ee5c6a805)
![image](https://github.com/MicrosoftDocs/powerapps-docs/assets/5404664/ee490d91-d2f9-4b49-af7b-73c9831d52b7)
